packages/server/src/api/controllers/static/index.ts

```diff
@@ -397,3 +528,74 @@ export const getSignedUploadURL = async function (
 
   ctx.body = { signedUrl, publicUrl }
 }
+
+export async function servePwaManifest(ctx: UserCtx<void, any>) {
+  const appId = context.getAppId()
+  if (!appId) {
+    ctx.status = 404
```

Comment:
These 3 lines could be simplified as `ctx.throw(404, message)`. Same on the other 404